scripts/generate_upgrade_syncset.py

```diff
@@ -0,0 +1,301 @@
+    Returns a list of clusters matching the supplied group and subgroup
+    :param group: Cluster group
+    :param subgroup: Cluster subgroup
+    :return: list of tuple pairs of cluster ID and cluster name
+    """
+
+    cluster_id_to_name = dict()
+
+    if cluster_ids is not None:
+        for cluster_id in cluster_ids:
+            try:
+                cluster_name = get_cluster_name_for_id(cluster_id, environment)
+                cluster_id_to_name[cluster_id] = cluster_name
+            except Exception as err:
+                logging.error('Unable to determine cluster name for cluster ID \'{}\': {}'.format(cluster_id, str(err)))
+                sys.exit(1)
+
+    if groups is not None:
+        for g in groups:
+            try:
+                # pout = subprocess.check_output(['cat', '/home/mbargenq/scratch/' + g + '.txt']).decode('utf-8')
+                pout = subprocess.check_output([CLUSTER_LIST_EXEC, g, subgroup]).decode('utf-8')
+                output_as_dict = json.loads(pout)
+
+                if 'items' not in output_as_dict:
+                    logging.error('Unexpected cluster info returned, unable to determine id/name')
+                    sys.exit(1)
+
+                cluster_items = output_as_dict['items']
+                for cluster in cluster_items:
+                    cluster_id_to_name[cluster['id']] = cluster['longName']
+
+            except OSError as err:
+                logging.info('Command \'{}\' failed with error: {}'.format(CLUSTER_LIST_EXEC, str(err)))
+                sys.exit(1)
+            except subprocess.CalledProcessError as err:
+                logging.info('Command \'{}\' failed with error: {}'.format(CLUSTER_LIST_EXEC, str(err)))
+                sys.exit(1)
+            except ValueError as err:
+                logging.info(
+                    'Output of command \'{}\' failed to parse with error: {}'.format(CLUSTER_LIST_EXEC, str(err)))
+                sys.exit(1)
+
+    return cluster_id_to_name
+
+
+def get_cluster_name_for_id(cluster_id, environment):
+    """
+    Determines the clusterdeployment name associated with the supplied clusterid
+    :param cluster_id: cluster-id of cluster
+    :return: clusterdeployment name associated with cluster ID
+    """
+    cluster_ns = 'uhc-{}-{}'.format(environment, cluster_id)
+    try:
+        # pout = subprocess.check_output(['cat', '/home/mbargenq/scratch/cluster.txt']).decode(
+        #     'utf-8')
+        pout = subprocess.check_output(['oc', 'get', 'clusterdeployment', '-n', cluster_ns, '-o', 'json']).decode(
+            'utf-8')
+        output_as_dict = json.loads(pout)
+        if 'items' not in output_as_dict:
+            raise Exception('fetching clusterdeployment returned invalid response')
+
+        cd = output_as_dict['items']
+        if len(cd) == 0:
+            raise Exception('clusterdeployment not found')
+
+        return cd[0]['metadata']['name']
+
+    except Exception as err:
+        raise err
```

Comment:
This is redundant. Just un-try/except the whole thing.

(Actually, to be pedantic, it's a teeny bit worse than redundant: by saying `raise err` rather than just `raise`, the exception that percolates up will have the stack trace from L100 rather than from wherever the actual exception occurred.)